src/persist/src/indexed/mod.rs

```diff
@@ -233,12 +233,20 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
 
         let mut updates: Vec<_> = updates
             .drain(..)
-            .map(|(_, (k, v), t, d)| ((k, v), t, d))
+            .map(|(_, (k, v), t, d)| (t, (k, v), d))
+            .collect();
+        // Future batches are required to be sorted and consolidated by ((ts, (k, v)).
+        differential_dataflow::consolidation::consolidate_updates(&mut updates);
+
+        if updates.is_empty() {
```

Comment:
nit: can move this above the consolidate_updates call

Reply:
I think you need to have it after because a batch might have a + 1 diff and -1 diff that consolidates to zero? you could also check for emptiness before the `consolidate_updates` call to avoid the function call, but i figured that's extremely inexpensive for empty arrays and not worth optimizing

Reply:
Oh great point! A little too late now but this would have been an excellent comment :)